### PR TITLE
UX: Disable button while searching.

### PR DESCRIPTION
**Pull request by tgxworld**
The issue is more prominent on mobile where the connection speed is slow.

### CODE REVIEW

File: app/assets/javascripts/discourse/controllers/full-page-search.js.es6 [hunk 1]
```diff
@@ -45,9 +46,9 @@ export default Ember.Controller.extend({
     return isValidSearchTerm(q);
   },
 
-  @computed('searchTerm')
-  isNotValidSearchTerm(searchTerm) {
-    return !isValidSearchTerm(searchTerm);
+  @computed('searchTerm', 'searching')
+  searchButtonDisabled(searchTerm, searching) {
+    return (searching || !isValidSearchTerm(searchTerm) ? true : false);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/full-page-search.js.es6]: Do you really need the ternary operator here?

Comment (SamSaffron) [inline app/assets/javascripts/discourse/controllers/full-page-search.js.es6]: also ... trick if you insist on api being pristine
```
!!(searching || !isValidSearchTerm(searchTerm))

or 

!searching && isValidSearchTerm(searchTerm)

```

:)

Comment (tgxworld) [inline app/assets/javascripts/discourse/controllers/full-page-search.js.es6]: > Do you really need the ternary operator here?

It was really late at night :stuck_out_tongue_closed_eyes: 



> also ... trick if you insist on api being pristine

:+1:

**Comment by SamSaffron** [comment]
sure ... sounds good!

**Pull request merged**